app/controllers/ticket_purchases_controller.rb

```diff
@@ -4,11 +4,11 @@ class TicketPurchasesController < ApplicationController
   authorize_resource :conference_registrations, class: Registration
 
   def create
+    TicketPurchase.destroy_all(user_id: current_user.id, conference_id: @conference.id, paid: false)
     message = TicketPurchase.purchase(@conference, current_user, params[:tickets][0])
     if message.blank?
-      if current_user.ticket_purchases.any?
-        redirect_to conference_conference_registration_path(@conference.short_title),
-                    notice: "Thank you for supporting #{@conference.title} by purchasing a ticket."
+      if current_user.ticket_purchases.unpaid.any?
+        redirect_to new_conference_payment_path, notice: 'Please pay here to purchase tickets.'
```

Comment:
Actually, line 7 deletes the unpaid ticket which failed their payments, then, new `ticket_purchases` are created on line 8 and then payment is attempted for new purchased_tickets on line 11